webui/src/apollo.ts

```diff
@@ -8,6 +8,25 @@ import introspectionQueryResultData from './fragmentTypes';
 
 const client = new ApolloClient({
   uri: '/graphql',
+  onError: ({ graphQLErrors, networkError }) => {
+    if (graphQLErrors) {
+      console.log(graphQLErrors);
+      for (const { message, locations, path, extensions } of graphQLErrors) {
+        if (extensions && extensions.resetStore) {
+          console.log(
+            'remote end returned error requesting that we reset the store'
+          );
+          client.resetStore();
+        }
```

Comment:
This does not seem necessary and may lead to request loops. If the server asks to reset the store for a query, the client would reset the store, and therefore re-run the same query, which would make a loop.

Reply:
Hmm. This is specifically to make sure the UI updates in response to being told "no, actually, you're not logged in"

Reply:
@lukegb is that a problem you saw with the current UI code or is it something you expect to be a problem in the future? As @sandhose said, this can trigger a request loop.

Reply:
It's mostly something I expect to be a problem once "proper" auth exists - I mostly ran into it while restarting the git-bug server between readonly and non-readonly mode, although this is a bit of an edge case.

Reply:
Ok, well, let's cross that bridge when we get there then.